core/scripts/chaincli/handler/handler.go

```diff
@@ -211,13 +220,54 @@ func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) erro
 	}
 
 	if receipt.Status == ethtypes.ReceiptStatusFailed {
+		// trying to extract failed reasons: https://ethereum.stackexchange.com/questions/48383/how-to-retrieve-revert-reason-for-past-transactions/80766#80766
+		reason, err := errorReason(ctx, h.client, h.fromAddr, tx, receipt.BlockNumber)
+		if err != nil {
+			log.Println("Failed to get error reason: ", err)
+		} else {
+			log.Println("Transaction failed. Reason: ", reason)
+		}
+
 		log.Println("Transaction failed: ", helpers.ExplorerLink(h.cfg.ChainID, tx.Hash()))
 		return errors.New("Transaction failed")
 	}
 
 	return nil
 }
 
+func errorReason(ctx context.Context, b ethereum.ContractCaller, from common.Address, tx *ethtypes.Transaction, blockNum *big.Int) (string, error) {
```

Comment:
I'm generally against having functions that don't belong to a particular type, especially when the function is only used by a single type - how would you feel about giving this function a `(h *baseHandler)` receiver? That way, you can eliminate the second and third arguments from the function signature, since the function will have access to the receiver properties?